nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -405,6 +405,14 @@ WaypointFollower::convertGPSPoses2MapPoses(
         "fromLL service of robot_localization could not convert %i th GPS waypoint to"
         "Map frame, going to skip this point!"
         "Make sure you have run navsat_transform_node of robot_localization");
+      if (stop_on_failure_) {
+        RCLCPP_ERROR(
+          parent_node->get_logger(),
+          "Conversion of %i th GPS waypoint to"
+          "Map frame failed and stop_on_failure is set to true"
+          "Not going to execute any of waypoints, exiting with failure!");
+        return std::vector<geometry_msgs::msg::PoseStamped>();
```

Comment:
Make sure that everywhere that uses this function can properly handle a 0 sized vector. The initial request will just return success if 0 length, but the preemption will  try to index at 0 which would cause a seg fault. Also I think that the action should return failure if there's 0 length on request. https://github.com/ros-planning/navigation2/blob/7c695202c3efaf7d9e79ed49582c52944ab002a7/nav2_waypoint_follower/src/waypoint_follower.cpp#L203